nixos/modules/services/misc/nzbget.nix

```diff
@@ -4,32 +4,34 @@ with lib;
 
 let
   cfg = config.services.nzbget;
-  dataDir = builtins.dirOf cfg.configFile;
-in {
-  options = {
-    services.nzbget = {
-      enable = mkEnableOption "NZBGet";
+  pkg = pkgs.nzbget;
+  stateDir = "/var/lib/nzbget";
```

Comment:
I'm not a big fan of hardcoding this path, I would prefer if we still had the dataDir option. Much because I tend to move things if I set things up... :-)

Reply:
Having ways to specify a custom `dataDir` complicates a lot of things, wrt. setting up these directories - we need to use `systemd.tmpfiles.rules` to set up these directories, instead of just `StateDirectory` - dynamic users are also problematic (see thread around https://github.com/NixOS/nixpkgs/pull/56265#discussion_r267738399).

If you want to have e.g. some files at a separate location, can't you just configure nzbget itself (or mount it inside `/var/lib/nzbget/…`?

Reply:
Hmm, ok. I wasn't familiar with how `StateDirectory` worked that you just give it the name of the directory that it should be named within `/var/lib/`.

If it wasn't for that it would have been more flexible. That makes sense but is still annoying :-)

Reply:
@etu Almost every other Linux distribution under the sun has standardized on a certain directory structure, which is reflected in the design of systemd. When I first started using NixOS I thought it was great that I could customize stuff like that (even though no other distro had this option...). I've since realized that by doing so we are fighting against systemd every step of the way. Like @flokli mentioned we should be working with systemd as opposed to against it. The value we offer our users by having such customizations which are hardly ever used (and which when are used can be achieved easily by other means, in this case symlinks) doesn't match the development effort.

I hope you will consider this, especially next time you write or review a module.